File: java/etl/src/main/java/Etl.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Etl {
  public Map<String, Integer> transform(Map<Integer, List<String>> old) {
    Map<String, Integer> letterToScore = new HashMap<>();
    for (int score : old.keySet()) {
      for (String letter : old.get(score)) {
        letterToScore.put(letter.toLowerCase(), score);
      }
    }

    return letterToScore;
  }
}

File: cpp/etl/etl.cpp
#include "etl.h"
#include <cctype>

namespace etl
{
std::map<char, int> transform(const std::map<int, std::vector<char>> &legacy_data)
{
    std::map<char, int> result;
    for (const auto &it : legacy_data)
    {
        for (char letter : it.second)
        {
            result[std::tolower(letter)] = it.first;
        }
    }

    return result;
}
} // namespace etl
